packages/page-experience/lib/checks/fontDisplay.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2021  The AMP HTML Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Checks whether a page uses the optimal font loading strategy for AMP pages. The recommemdation is to
+ * use `font-display: optional` for critical fonts used in the first viewport. This improves LCP on slow
+ * connections and avoids content-shifts resulting from font swapping.
+ *
+ * @param {Object} pageData
+ */
+const checkDisplayOptional = (pageData, result) => {
+  for (const font of pageData.criticalFonts) {
+    const fontface = pageData.fontFaces.get(font);
+    if (!fontface) {
+      continue;
+    }
+
+    if (!fontface.fontDisplay) {
+      result.warn({
+        title: 'Improve LCP using `font-display: optional`',
```

Comment:
Would it make sense to extract the warnings into an external file?

Reply:
For translation?